File: ProyectoCsharp/Modelo/Categoria.cs
using System;
using System.Collections.Generic;

namespace Modelo
{
    public class Categoria
    {
        private long id_categoria;
        private string nombre;
        private string descripcion;
        private List<Articulo> articulos;



        //Constructores
        public Categoria()
        {
            id_categoria = 0;
            nombre = descripcion = null;
            articulos = new List<Articulo>();
        }

        public Categoria(long id_categoria, string nombre, string descripcion)
        {
            Id_categoria = id_categoria;
            Nombre = nombre;
            Descripcion = descripcion;
            articulos = new List<Articulo>();
        }

        //Destructor
        ~Categoria()
        {
            nombre = descripcion = null;
            articulos = null;
        }

        //Metodos set y get
        public long Id_categoria { get => id_categoria; set => id_categoria = value; }
        public string Nombre { get => nombre; set => nombre = value; }
        public string Descripcion { get => descripcion; set => descripcion = value; }

        public void AgregarArticulo(Articulo articulos)
        {
            this.articulos.Add(articulos);
        }

        public List<Articulo> GetArticulos()
        {
            return this.articulos;
        }

        public override string ToString()
        {
            return String.Format("{0}", Nombre);
        }

    }
}

File: TrabajoJavaLp2/src/modelo/Categoria.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modelo;

import java.util.ArrayList;

/**
 *
 * @author Lenovo
 */
public class Categoria {
    private long id_categoria;
	private String nombre;
	private String descripcion;
	private ArrayList<Articulo> articulos;
	public Categoria(){
		id_categoria = 0;
		nombre = null;
		descripcion = null;
		articulos = new ArrayList<Articulo>();
	}

	public Categoria(long id_categoria, String nombre, String descripcion){
		setId_categoria(id_categoria);
		setNombre(nombre);
		setDescripcion(descripcion);
	}

	public void setId_categoria(long id_categoria){
		this.id_categoria = id_categoria;
	}

	public long getId_categoria(){
		return id_categoria;
	}

	public void setNombre(String nombre){
		this.nombre = nombre;
	}

	public String getNombre(){
		return nombre;
	}

	public void setDescripcion(String descripcion){
		this.descripcion = descripcion;
	}

	public String getDescripcion(){
		return descripcion;
	}
	public void AgregarArticulo(Articulo articulos){
		this.articulos.add(articulos);
	}
	
	public ArrayList<Articulo> articulos(){
		return this.articulos;
	}
}
